NNG asserts when setting socket name

## NNG & Platform details.

NNG version: v1.0.1
Operating System: Linux
Language: C
CMake build type: Debug

## Expected Behavior

The following sample should run without crashing.

```C
#include <nng/nng.h>                                                                                                                                                                          
#include <nng/protocol/reqrep0/rep.h>

int main()
{
  int status;

  // Create socket
  nng_socket sock;
  status = nng_rep0_open(&sock);
  if (status) return status;

  // Set socket name
  status = nng_setopt_string(sock, "socket-name", "test-socket");
  if (status) return status;

  // Close socket
  status = nng_close(sock);
  return status;
}
```

## Actual Behavior

When NNG is built without `-DNDEBUG` the above sample outputs

```
[thom@laptop nng_bug]$ ./test 
panic: /home/thom/Programs/slamcore_deps/deps/nng/src/core/options.c: 161: assert err: sz == z
This message is indicative of a BUG.
Report this at http://github.com/nanomsg/nanomsg                                                                                                                                              
/usr/local/lib/libnng.so.1(+0x168aa) [0x7ff985d7a8aa]                                                                                                                                         
/usr/local/lib/libnng.so.1(+0x15dfa) [0x7ff985d79dfa]
/usr/local/lib/libnng.so.1(+0x187b3) [0x7ff985d7c7b3]
/usr/local/lib/libnng.so.1(+0x1a15e) [0x7ff985d7e15e]
/usr/local/lib/libnng.so.1(+0xd889) [0x7ff985d71889]
/usr/local/lib/libnng.so.1(nng_setopt_string+0x3b) [0x7ff985d71b23]
./test(+0x11b2) [0x555cd19f61b2]
/usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7ff985bc4223]
./test(+0x109e) [0x555cd19f609e]
Aborted (core dumped)
```

## Steps to Reproduce

Install NNG from source and pass `-DCMAKE_BUILD_TYPE=Debug` when running cmake.

## Suggested patch

```diff
diff --git a/src/core/options.c b/src/core/options.c
index 3e35f44c..ea0daafd 100644
--- a/src/core/options.c
+++ b/src/core/options.c
@@ -157,9 +157,6 @@ nni_copyin_str(char *s, const void *v, size_t sz, size_t maxsz, nni_opt_type t)
 
        switch (t) {
        case NNI_TYPE_STRING:
-               z = strlen(v) + 1;
-               NNI_ASSERT(sz == z);
-               break;
        case NNI_TYPE_OPAQUE:
                if ((z = nni_strnlen(v, sz)) >= sz) {
                        return (NNG_EINVAL); // missing terminator
```

--- a/src/core/options.c
+++ b/src/core/options.c
@@ -157,9 +157,6 @@ nni_copyin_str(char *s, const void *v, size_t sz, size_t maxsz, nni_opt_type t)
 
 	switch (t) {
 	case NNI_TYPE_STRING:
-		z = strlen(v) + 1;
-		NNI_ASSERT(sz == z);
-		break;
 	case NNI_TYPE_OPAQUE:
 		if ((z = nni_strnlen(v, sz)) >= sz) {
 			return (NNG_EINVAL); // missing terminator

--- a/tests/sock.c
+++ b/tests/sock.c
@@ -122,6 +122,27 @@ TestMain("Socket Operations", {
 				nng_strfree(allocd);
 			});
 
+			Convey("Oversize sockname handled right", {
+				char   name[256]; // 64 is max
+				size_t sz = sizeof(name);
+				memset(name, 'A', sz);
+				So(nng_setopt(s1, NNG_OPT_SOCKNAME, name,
+				       sz) == NNG_EINVAL);
+				name[sz - 1] = '\0';
+				So(nng_setopt(s1, NNG_OPT_SOCKNAME, name,
+				       sz) == NNG_EINVAL);
+
+				strcpy(name, "hello");
+				So(nng_setopt(
+				       s1, NNG_OPT_SOCKNAME, name, sz) == 0);
+				sz = sizeof(name);
+				memset(name, 'B', sz);
+				So(nng_getopt(
+				       s1, NNG_OPT_SOCKNAME, name, &sz) == 0);
+				So(sz == 6);
+				So(strcmp(name, "hello") == 0);
+			});
+
 			Convey("RAW option works", {
 				bool raw;
 				So(nng_getopt_bool(s1, NNG_OPT_RAW, &raw) ==

